community/lucene-index/src/main/java/org/neo4j/kernel/api/impl/index/AbstractLuceneIndex.java

```diff
@@ -56,10 +58,9 @@
  */
 public abstract class AbstractLuceneIndex
 {
-
     protected final PartitionedIndexStorage indexStorage;
     private final IndexPartitionFactory partitionFactory;
-    private List<AbstractIndexPartition> partitions = new CopyOnWriteArrayList<>();
+    private CopyOnWriteArrayList<AbstractIndexPartition> partitions = new CopyOnWriteArrayList<>();
```

Comment:
I changed it because we are now relying on specifics of the implementation, even if we are not using any methods not already defined in `List`. Giving the field a concrete type helps emphasise this, but I can also add a comment. What do you think?